### PR TITLE
Added WAF vendors headers in traces tags

### PR DESCRIPTION
**Pull request by vpellan**
**What does this PR do?**
Add WAF vendors headers in traces tags according to this [RFC](https://docs.google.com/document/d/1xf-s6PtSr6heZxmO_QLUtcFzY_X_rT94lRXNq6-Ghws)

**Motivation:**
This will enable the ASM Respond team to parse the value of these headers and check that they are in the correct format

**Additional Notes:**
I used a hash to access traces-formatted headers and rack-formatted headers at the same time, and this hash is created from a list of string, corresponding to the HTTP-formatted header. Maybe these operations can add overhead time, and manually creating the hash would be faster

**How to test the change?**

1. Use Rspec (`-e 'with WAF vendor headers'`)
2. Use [system-tests](https://github.com/DataDog/system-tests)
	- Enable Test_ExternalWafRequestsIdentification in ruby manifest
	- `run.sh tests/appsec/test_traces.py::Test_ExternalWafRequestsIdentification::test_external_wafs_header_collection`
	- Check on datad0g staging that the trace contains WAF headers tags

**For Datadog employees:**
- [ ] If this PR touches code that signs or publishes builds or packages, or handles
credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

**JIRA**
- [APPSEC-52081](https://datadoghq.atlassian.net/browse/APPSEC-52081)
- [APPSEC-52097](https://datadoghq.atlassian.net/browse/APPSEC-52097)

[APPSEC-52081]: https://datadoghq.atlassian.net/browse/APPSEC-52081?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/datadog/appsec/contrib/rack/request_middleware.rb [hunk 1]
```diff
@@ -54,7 +66,8 @@ def call(env)
 
             gateway_request = Gateway::Request.new(env)
 
-            add_appsec_tags(processor, scope, env)
+            add_appsec_tags(processor, scope)
+            add_http_headers_tags(scope, env)
```
Comment (lloeki) [inline lib/datadog/appsec/contrib/rack/request_middleware.rb:70]: See below

```suggestion
            add_request_tags(scope, env)
```

**Review by lloeki — APPROVED** [review]
LGTM, save for a minor suggestion

**Review by ivoanjo — CHANGES_REQUESTED** [review]
Note: Please hold off on merging this to master until we get the 1.21.1 release out. (Feel free to dismiss afterwards! Thanks for the patience!)

File: lib/datadog/appsec/contrib/rack/request_middleware.rb [hunk 2]
```diff
@@ -13,13 +13,28 @@ module Datadog
...
+          Cloudfront-Viewer-Ja3-Fingerprint
+          Cf-Ray
+          X-Cloud-Trace-Context
+          X-Appgw-Trace-id
+          X-SigSci-RequestID
+          X-SigSci-Tags
+          Akamai-User-Risk
+        ].map(&:downcase).freeze
+
+        # .map { |s| [s.downcase, Datadog::Tracing::Contrib::Rack::Header.to_rack_header(s)] }.to_h
```
Comment (lloeki) [inline lib/datadog/appsec/contrib/rack/request_middleware.rb:28]: Leftover comment, to be removed ;)

**Review by lloeki — APPROVED** [review]
LGTM, just an extra comment left

File: lib/datadog/appsec/contrib/rack/request_middleware.rb [hunk 3]
```diff
@@ -13,13 +13,26 @@ module Datadog
   module AppSec
     module Contrib
       module Rack
+        # Create an array of lowercased headers
```
Comment (ivoanjo) [inline lib/datadog/appsec/contrib/rack/request_middleware.rb:16]: Minor: I don't think this comment adds a lot, reminds me of this meme lol :wink: :wink: :wink: :wink::

**Pull request merged**